### PR TITLE
Bring over blues code from other projects.

### PR DESCRIPTION
**Pull request by jcsackett**

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/theblues-lib/82/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/theblues-lib/83/
Test PASSed.

**Comment by jcsackett** [comment]
Note:

To get theblues code merged into one repository things are temporarily a bit jenky--the JEM code doesn't really work in a standalone form, and py34 support had to be disabled until things can work w/o macaroons (which will be my next branch).

File: Makefile [hunk 1]
```diff
@@ -98,7 +98,8 @@ test-all: $(TOX)
 	$(TOX)
 
 .PHONY: check
-check: clean test-all
+#check: clean test-all
```
Comment (bac) [inline Makefile:101]: Do we still need this comment?

File: theblues/charmstore.py [hunk 2]
```diff
@@ -1,4 +1,8 @@
 import logging
+try:
```
Comment (bac) [inline theblues/charmstore.py:2]: Please comment why this machination is needed.  Py 2 / 3 issue?
Oops, I see this is just moved code.

File: Makefile (hunk 1 above)
Comment (makyo) [inline Makefile]: Needed?

File: tox.ini [hunk 3]
```diff
@@ -2,6 +2,10 @@
 envlist = py27, py34, style, docs
 
 [testenv]
+# XXX j.c.sackett 2016-04-07 Long term we want to remote this directive,
```
Comment (makyo) [inline tox.ini]: s/remote/remove

**Comment by makyo** [comment]
:+1: QA (`make check`) :ok:

File: Makefile (hunk 1 above)
Comment (jcsackett) [inline Makefile:101]: It's presence makes it a great deal easier to work on getting test-all working again, so yes.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/theblues-lib/84/
Test PASSed.

**Comment by bac** [comment]
:+1:

**Comment by jcsackett** [comment]
Thanks for the reviews, all. :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/theblues-lib-merge

**Pull request merged**